SonarQube.Common/Utilities.cs

```diff
@@ -213,6 +218,80 @@ public static void SafeDispose(object instance)
             }
         }
 
+        /// <summary>
+        /// Returns true if the given SARIF came from the VS 2015 RTM Roslyn, which does not provide correct output.
+        /// </summary>
+        public static bool IsSarifFromUnsupportedCompiler(string input)
+        {
+            // low risk of false positives / false negatives
+            if (input.Contains(@"""toolName"": ""Microsoft (R) Visual C# Compiler""")
+                && input.Contains(@"""productVersion"": ""1.0.0"""))
+            {
+                return true;
+            }
+
+            // all other cases
+            return false;
+        }
+        
+        /// <summary>
+        /// Used to correct invalid SARIF generated by VS 2015 RTM Roslyn. 
+        /// Replaces all slashes in any line beginning with "uri": with double slashes.
+        /// Applying this to already-valid SARIF will result in over-escaping.
+        /// </summary>
+        public static bool FixImproperlyEscapedSarif(string input, out string output)
+        {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Consider a design that does not require that 'output' be an out parameter. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=fxcop%3AAvoidOutParameters)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Correct the spelling of 'Sarif' in member name 'Utilities.FixImproperlyEscapedSarif(string, out string)' or remove it entirely if it represents any sort of Hungarian notation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=fxcop%3AIdentifiersShouldBeSpelledCorrectly)